src/errors/error-handler.js

```diff
@@ -0,0 +1,137 @@
+const { ObjectId } = require('mongodb');
+const XDate = require("xdate");
+
+
+//These functions are meant to be used in if statements, so that you can make your own throw statement.
+
+function checkObjectId(OId){
+    if(ObjectId.isValid(OId)){
+        if((String)(new ObjectId(OId)) == OId){
+            return true;        
+        }
+        return false;
+    }
+    return false;
+}
+//checks if input is a string
+function checkString(str){
+    return (typeof(str) === 'string');
+}
+
+//checks if input is a number
+function checkNumber(num){
+    return (typeof(num) === 'number'); 
+}
+
+function checkStoryPoint(sp){
+    return (checkNumber(sp) && sp > 0 && sp < 6);
+}
+
+//REGEX is for the current email format RFC2822
+function checkEmail(email){
+    return (typeof(email) === 'string' && /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/.test(email))
+}
+
+//This will not work if spaces are at the end, however spaces should not be inbetween letters 
+//just make sure to trim the end of the string before using this function 
+function checkFirstName(firstName){
+    return(typeof(firstName) === 'string' && /^[a-zA-Z]+$/.test(firstName));
+}
+
+//Checks if input is a valid last name
+function checkLastName(lastName){
+    return(typeof(lastName) === 'string' && /^[a-zA-Z'-]+$/.test(lastName));
+}
+
+//Checks if input is a valid hex color code
+function checkColor(colorCode){
+    return (typeof(colorCode) === 'string' && /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/.test(colorCode));
+}
+
+//Checks if input is a valid array of objectid's
+function checkArrayObjectId(boardArray){
+    if(!Array.isArray(boardArray)){
+        console.log('here')
+        return false;
+    }
+    for(let x of boardArray){
+        if(!checkObjectId(x)){
+            return false;
+        }
+    }
+    return true;
+}
+
+
+//Checks if input is a valid date in the form MM/DD/YYYY
+function checkDate(dateStr){
```

Comment:
Didn't this date code not work when he tested it? I have robust date code from a previous assignment we can use here